scripts/o3de/o3de/engine_template.py

```diff
@@ -1543,21 +1608,19 @@ def create_project(project_path: pathlib.Path,
 
     # project restricted name
     if project_restricted_name and not project_restricted_path:
-        project_restricted_path = manifest.get_registered(restricted_name=project_restricted_name)
+        gem_restricted_path = manifest.get_registered(restricted_name=project_restricted_name)
+        if not gem_restricted_path:
+            logger.error(f'Project Restricted Name {project_restricted_name} cannot be found.')
+            return 1
 
     # project restricted path
-    elif project_restricted_path:
+    if project_restricted_path:
         if not os.path.isabs(project_restricted_path):
-            default_projects_restricted_folder = manifest.get_registered(restricted_name='projects')
-            new_project_restricted_path = default_projects_restricted_folder/ project_restricted_path
-            logger.info(f'Project restricted path {project_restricted_path} is not a full path, we must assume its'
-                        f' relative to default projects restricted path = {new_project_restricted_path}')
-            project_restricted_path = new_project_restricted_path
-    elif template_restricted_path:
-        project_restricted_default_path = manifest.get_registered(restricted_name='projects')
-        logger.info(f'--project-restricted-path is not specified, using default project restricted path / project name'
-                    f' = {project_restricted_default_path}')
-        project_restricted_path = project_restricted_default_path
+            logger.error(f'Project Restricted Path {project_restricted_path} is not an absolute path.')
+            return 1
+    # neither put it in the default restricted projects
+    else:
+        project_restricted_path = manifest.get_o3de_restricted_folder() / 'Projects' / project_name
```

Comment:
This will cause project creation to fail if it has a "restricted" key that doesn't have a mapping in registered in the o3de_manifest.json.
A user without a restricted directory will find that project creation will now fail for them.